lib/srv/discovery/discovery.go

```diff
@@ -403,29 +415,43 @@ func (s *Server) handleEC2Instances(instances *server.EC2Instances) error {
 	return trace.Wrap(s.ec2Installer.Run(s.ctx, req))
 }
 
+func (s *Server) logHandleInstancesErr(err error) {
+	var aErr awserr.Error
+	if errors.As(err, &aErr) && aErr.Code() == ssm.ErrCodeInvalidInstanceId {
+		s.Log.WithError(err).Error("SSM SendCommand failed with ErrCodeInvalidInstanceId. Make sure that the instances have AmazonSSMManagedInstanceCore policy assigned. Also check that SSM agent is running and registered with the SSM endpoint on that instance and try restarting or reinstalling it in case of issues. See https://docs.aws.amazon.com/systems-manager/latest/APIReference/API_SendCommand.html#API_SendCommand_Errors for more details.")
+	} else if trace.IsNotFound(err) {
+		s.Log.Debug("All discovered EC2 instances are already part of the cluster.")
+	} else {
+		s.Log.WithError(err).Error("Failed to enroll discovered EC2 instances.")
+	}
+}
+
 func (s *Server) handleEC2Discovery() {
 	if err := s.nodeWatcher.WaitInitialization(); err != nil {
 		s.Log.WithError(err).Error("Failed to initialize nodeWatcher.")
 		return
 	}
 
 	go s.ec2Watcher.Run()
+	go s.agentlessEC2Watcher.Run()
+	go s.watchCARotations(s.ctx)
+
 	for {
 		select {
+		case instances := <-s.agentlessEC2Watcher.InstancesC:
```

Comment:
> should we just blindly re-run the teleport install process on all ec2 instances if there has been a rotation?

This sounds good to me, any reason it wouldn't work? For instances that miss rotation stages, can we just encode the current CA serial number or something in the node resource labels which would they sync back when running `openssh join` command? Then the reconciler can detect instances that are running an incorrect CA and re-run joining on them.

> return to the previous plan of installing a systemd timer (or a cron job if systemd isnt available?) that tries to rotate the certs every x hours.

I am not a fan of this approach because it basically becomes not agentless anymore. Who's gonna upgrade this systemd unit, how do we detect that it broke or stopped, etc. Then we're basically to square one, at that point you as well may just run Teleport process on the nodes.